lib/mayaUsd/fileio/utils/meshWriteUtils.cpp

```diff
@@ -377,4 +405,76 @@ UsdMayaMeshUtil::exportReferenceMesh(UsdGeomMesh& primSchema, MObject obj)
     primVar.GetAttr().Set(VtValue(points));
 }
 
+void
+UsdMayaMeshUtil::assignSubDivTagsToUSDPrim(MFnMesh& meshFn,
+                                           UsdGeomMesh& primSchema,
+                                           UsdUtilsSparseValueWriter& valueWriter)
+{
+    // Vert Creasing
+    MUintArray mayaCreaseVertIds;
+    MDoubleArray mayaCreaseVertValues;
+    meshFn.getCreaseVertices(mayaCreaseVertIds, mayaCreaseVertValues);
+    if (!TF_VERIFY(mayaCreaseVertIds.length() == mayaCreaseVertValues.length())) {
+        return;
+    }
+    if (mayaCreaseVertIds.length() > 0u) {
+        VtIntArray subdCornerIndices(mayaCreaseVertIds.length());
+        VtFloatArray subdCornerSharpnesses(mayaCreaseVertIds.length());
+        for (unsigned int i = 0u; i < mayaCreaseVertIds.length(); ++i) {
+            subdCornerIndices[i] = mayaCreaseVertIds[i];
+            subdCornerSharpnesses[i] = mayaCreaseVertValues[i];
+        }
```

Comment:
Add a little inline utility method.... 
```c++
inline void doubleToFloat(float* const f, const double* const d, size_t c)
{
  for(size_t i = 0; i < c; ++i)
    f[i] = d[i];
}
```
and refactor to:
```c++
        VtIntArray subdCornerIndices(mayaCreaseVertIds.length());
        VtFloatArray subdCornerSharpnesses(mayaCreaseVertIds.length());
        memcpy(subdCornerIndices.data(), &mayaCreaseVertIds[0], mayaCreaseVertIds.length() * sizeof(int));
        doubleToFloat(subdCornerSharpnesses.data(), &mayaCreaseVertValues[0], mayaCreaseVertValues.length());
```
As for the reason why, take a look here:
https://godbolt.org/z/etT0v_

The above loop is effectively doing the same as copyA. If you look at the ASM, you'll noticce that all of the double->float conversions are being done via vcvtsd2ss (i.e. 1 value at a time). If however you use a dumb inline function (i.e. copyB), you'll notice that the generated ASM uses vcvtpd2ps (convert 4 at a time), no longer needs to make any function calls within the loop, and should make this method a wee bit faster allround...

Reply:
@robthebloke  Good point and thanks for the godbolt example. 

Can I ask you if we can make this change in another PR? the reason I am asking is because the suggested `doubleToFloat`  utility functions is already part of `plugin\al\usdmayautils\AL\usdmaya\utils\MeshUtils.cpp`  and I already have a plan to bring some of these utilities to the core library so others can use them.  This is certainly a work for another PR.